healthcheck/healthcheck.go

```diff
@@ -61,15 +70,21 @@ func newTickers(interval time.Duration, clients []*Client) []*ticker {
 	return tickers
 }
 
-// start begins each ticker, this is used to run the health checks on dependent apps
-func (hc HealthCheck) start(ctx context.Context) {
+// Start begins each ticker, this is used to run the health checks on dependent apps
+// takes argument context and should utilise contextWithCancel
+func (hc *HealthCheck) Start(ctx context.Context) {
```

Comment:
Context is generally passed as a pointer, though I'm not sure it matters a huge amount